pkg/apis/wksprovider/machine/os/os.go

```diff
@@ -59,8 +59,10 @@ const (
 var (
 	pemKeys            = []string{"certificate-authority", "client-certificate", "client-key"}
 	fluxSecretTemplate = `apiVersion: v1
+{{ if .SecretValue }}
```

Comment:
this raised a red flag to me because I was unsure whether flux allows the total absence of this value, but your commit description added all the necessary context.  very helpful.  I wasn't sure if it would be better to wrap the `if` statement around just the `identity` section (rather than `data.identity` as it is now) partly because that's more semantically meaningful to the actual intent, but also because I wasn't sure if omitting `data` is allowed by the API.

I looked up the [OpenAPI spec](https://raw.githubusercontent.com/kubernetes/kubernetes/master/api/openapi-spec/swagger.json) for secrets and can confirm it isn't necessary.  I'll therefore leave it to you to decide if you want to adjust the if statement to cover just the identity part.  That said... I believe that flux may itself not add the data field when initially creating the secret but I couldn't find the code to prove it.